llm_guard/input_scanners/ban_competitors.py

```diff
@@ -68,11 +73,13 @@ def __init__(
             "ner", model=tf_model, tokenizer=tf_tokenizer, **model.pipeline_kwargs
         )
 
-    def scan(self, prompt: str) -> (str, bool, float):
+    def scan(self, prompt: str) -> tuple[str, bool, float]:
         is_detected = False
         text_replace_builder = TextReplaceBuilder(original_text=prompt)
         entities = self._ner_pipeline(prompt)
+        assert isinstance(entities, list)
```

Comment:
We had a similar discussion about the usage of assert in the code https://github.com/protectai/llm-guard/issues/71

Reply:
In this instance assert is only used for [type narrowing](https://mypy.readthedocs.io/en/stable/type_narrowing.html)

`entities` type  can be one of the following: list[Unknown] | list[Any] | PipelineIterator | Generator[Any, Any, None] | Tensor | Unknown | Any | None

<img width="464" alt="Screenshot 2024-05-22 at 3 47 03 PM" src="https://github.com/protectai/llm-guard/assets/8380706/fd464cdc-8823-418d-a8c7-a2422dad910c">

sorted is expecting an iterable or a list.  To narrow the type and force the type checker that a broader type is actually more specific, in this instance a list, we are using assert in conjunction with isinstance.

Without this we get the following error.
<img width="556" alt="Screenshot 2024-05-22 at 3 51 18 PM" src="https://github.com/protectai/llm-guard/assets/8380706/9812b077-b13f-4916-9a36-2c767ea1b15b">.

Here we aren't using assert for debugging or to throw an error.  It is okay if this is removed if compiling for optimization.